src/components/DocumentManager/hooks/DocumentManagerState.js

```diff
@@ -6,35 +6,41 @@ import {
 
 function useDocumentManagerState(collectionId) {
   let [documents, setDocuments] = useState(null);
+  // TODO: Suggest storing only the selectedDocument's id here. Doing so avoids having the full data in two spots, which reduces memory usage and avoids the risk of out of sync
+  // between the two datasets.
   let [selectedDocument, setSelectedDocument] = useState(null);
 
   useEffect(() => {
     (async () => {
-      const collectionDocuments = await getCollectionDocuments(collectionId);
-      setDocuments(collectionDocuments);
-
-      if (collectionDocuments.length > 0) {
-        const documentWithFile = await getDocument(collectionDocuments[0].id);
-        updateCollectionDocument(collectionDocuments, documentWithFile);
-        setSelectedDocument(documentWithFile);
+      try {
+        const docs = await getCollectionDocuments(collectionId);
+        if (docs.length > 0) {
+          const doc = await getDocument(docs[0].id);
+          updateState(docs, doc);
+        }
+      } catch (error) {
+        alert("Loading documents failed. Please try reloading your browser.");
       }
     })();
-  }, []);
+  }, [collectionId]);
 
-  function updateCollectionDocument(collection, document) {
-    const index = collection.findIndex(d => d.id === document.id);
-    collection[index] = document;
+  function updateState(docs, doc) {
+    setDocuments(docs.map(d => (d.id === doc.id ? doc : d)));
+    setSelectedDocument(doc);
```

Comment:
I eliminated the state mutation. React state should be treated as immutable. Instead, using map which returns a new array and weaving the changed record in. Also moved `setSelectedDocument` here to simplify call sites since they both need to do that too.

Reply:
Fixed